### ISSUE TITLE
Adobe Pass as Kodi Script

### ISSUE THREAD

**Issue by eracknaphobia** [awaters1/plugin.video.espn_3#3]
Any interest in working together on building an Adobe Pass kodi script which would enable developers to simplify the login process. I started a [repo](https://github.com/eracknaphobia/script.module.adobepass) a while back but never really got anywhere with it beyond a  Simpson World addon I made for myself. It appears you have made some big strides with the espn 3 addon and it would be great to bring some of that functionality into one central script.

**Comment by awaters1** [comment]
Yes I agree this would be a good path. 

I borrowed a lot of your implementation but I was not a fan of having a separate routine for each provider so I tried to simulate what the browser would do be doing the following
1. User chooses a provider (MSO) within settings, each provider is assigned a unique ID that adobe handles (http://api-app.espn.com/v1/watch/clients/watchespn-flash/providers?_accept=text/xml&mvpd=true the abbreviation tag)
2. This gets plugged into the IDP url (https://sp.auth.adobe.com/adobe-services/1.0/authenticate)
3. When the IDP url loads it does one of two things
3a. Uses an HTTP equiv redirect header to send the user to their providers page
3b. Presents an HTTP POST form to send the user to their providers page
This steps basically passes the SAMLRequest to the provider and presents the user with the login page
4. Sniff out the username and password field using heuristics (right now all text is populated with the username and all password are populated with the password) also sniff out the POST url
5. Post the response to the POST url in 4. and read the response
6. Each response should contain the SAMLResponse that is used to continue the rest of the adobe process

But I think that the provider specific scripts that you worked on should be implemented as some sort of fallback method, either that or the generic steps I describe above can be used as a fallback method when a specific provider script doesn't exist.

I will fork the above repository and make a branch in this repository to test out the script dependency.

One thing that is missing from this addon is the maintainSession logic.  I think the streams are supposed to stop if there isn't a ping back to espn's authentication system, but I'm not sure.
